test/sqllogictest/with_mutually_recursive.slt

```diff
@@ -158,21 +158,21 @@ SELECT (SELECT COUNT(*) FROM foo) FROM bar;
 ## Test error cases
 
 ## Test a recursive query with mismatched types.
-statement error did not match inferred type
+statement error db error: ERROR: WITH MUTUALLY RECURSIVE query "bar" declared types \(integer\), but query returns types \(text\)
 WITH MUTUALLY RECURSIVE
     foo (a text, b int) AS (SELECT 1, 2 UNION SELECT a, 7 FROM bar),
     bar (a int) as (SELECT a FROM foo)
 SELECT * FROM bar;
 
 ## Test with fewer columns than declared
-statement error did not match inferred type
+statement error db error: ERROR: WITH MUTUALLY RECURSIVE query "foo" declared types \(integer, integer\), but query returns types \(integer\)
 WITH MUTUALLY RECURSIVE
     foo (a int, b int) AS (SELECT 1 UNION SELECT a FROM bar),
     bar (a int) as (SELECT a FROM foo)
 SELECT a FROM foo, bar;
 
 ## Test with more columns than declared
-statement error did not match inferred type
+statement error db error: ERROR: WITH MUTUALLY RECURSIVE query "foo" declared types \(integer, integer\), but query returns types \(integer, integer, integer\)
```

Comment:
Does this show typmods properly? I know the whole inspiration behind this work is that numeric typmods are now automatically cast, but I'm curious whether providing an integer when, say, a `numeric(1, 2)` is required properly renders the `numeric(1, 2)`.

Reply:
We don't include the typmod, which is a feature/limitation of `humanize_scalar_type`––it only refers to the type objects. With the assignment-type casts in place, this makes sense because the errors have to be more dramatic than the typmods being mismatched (i.e. you gave us text and we needed a number). So including the typmod won't give folks meaningful context to fix the problem.